src/code_bevy_release/examples/pan-orbit-camera.rs

```diff
@@ -2,81 +2,141 @@ use bevy::prelude::*;
-            camera.focus += translation;
-            translation.z = -scroll;
-            trans.translation += translation;
+            any = true;
+
+            let window = get_primary_window_size(&windows);
+            let delta_x = {
+                let delta = rotation_move.x / window.x * std::f32::consts::PI * 2.0;
+                if pan_orbit.upside_down { -delta } else { delta }
+            };
+            let delta_y = rotation_move.y / window.y * std::f32::consts::PI;
+            let yaw = Quat::from_rotation_y(-delta_x);
+            let pitch = Quat::from_rotation_x(-delta_y);
+            pan_orbit.rotation = yaw * pan_orbit.rotation; // rotate around global y axis
+            pan_orbit.rotation = pan_orbit.rotation * pitch; // rotate around local x axis
+        } else if pan.length_squared() > 0.0 {
+            any = true;
+            // make panning distance independent of resolution,
+            let window = get_primary_window_size(&windows);
+            let ref_height = 0.829;
+            let reference = Vec2::new(ref_height * window.x / window.y, ref_height);
+            let norm = reference / window;
+            pan *= norm;
+            // translate by local axes
+            let right = transform.rotation * Vec3::unit_x() * -pan.x;
+            let up = transform.rotation * Vec3::unit_y() * pan.y;
+            // make panning proportional to distance away from focus point
+            let translation = (right + up) * pan_orbit.zoom;
+            pan_orbit.focus += translation;
+        } else if scroll.abs() > 0.0 {
+            any = true;
+            pan_orbit.zoom -= scroll * pan_orbit.zoom * 0.2;
+            // dont allow zoom to reach zero or you get stuck
+            pan_orbit.zoom = f32::max(pan_orbit.zoom, 0.05);
+        }
+
+        if any {
+            // emulating parent/child to make the yaw/y-axis rotation behave like a turntable
+            // parent = x and y rotation
+            // child = z-offset
+            let parent = Mat4::from_rotation_translation(pan_orbit.rotation, pan_orbit.focus);
+            let child = Mat4::from_rotation_translation(Quat::identity(), Vec3::new(0.0, 0.0, pan_orbit.zoom));
+            let (_, rot, pos) = parent.mul_mat4(&child).to_scale_rotation_translation();
+            transform.translation = pos;
+            transform.rotation = rot;
         }
     }
 }
 
-/// Spawn a camera like this.
+fn get_primary_window_size(windows: &Res<Windows>) -> Vec2 {
+    let window = windows.get_primary().unwrap();
+    let window = Vec2::new(window.width() as f32, window.height() as f32);
+    window
+}
+
+/// Spawn a camera like this
 fn spawn_camera(commands: &mut Commands) {
-    commands.spawn(Camera3dBundle::default())
-        .with(PanOrbitCamera::default());
+    use bevy::render::camera::PerspectiveProjection;
+
+    let camera = PanOrbitCamera::default();
+    let zoom = camera.zoom;
```

Comment:
I don't think the "zoom" field should be called that. This is not camera zoom. It's confusing.

"Zoom" in 3D Perspective cameras is a change of the projection's FOV.

For visually-correct zoom (like an actual camera behaves), you reduce the FOV to zoom in (telephoto), or increase the FOV to zoom out (wide-angle). The camera stays positioned in the same place.

Your "zoom" field does not seem to refer to that, but rather to the radius (distance from center / `focus`) of orbit. This is correct for our control scheme, since the scroll wheel is used to move closer to or farther away.

I think you should rename `zoom` to `radius` everywhere, just to avoid confusion and overloading the term.